store/store.go

```diff
@@ -0,0 +1,322 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package store
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"gopkg.in/macaroon.v1"
+
+	"github.com/CanonicalLtd/serial-vault/datastore"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/overlord/auth"
+)
+
+const (
+	ssoBaseURL   = "https://login.ubuntu.com/api/v2/"
+	storeBaseURL = "https://dashboard.snapcraft.io/dev/api/"
+)
+
+// Permissions is the SSO authorization for the store
+type Permissions struct {
+	Permissions []string `json:"permissions"`
+}
+
+// ACL is the SSO authorization for the store
+type ACL struct {
+	Macaroon string `json:"macaroon"`
+}
+
+// Discharge is the SSO authorization for the store
+type Discharge struct {
+	Macaroon string `json:"discharge_macaroon"`
+}
+
+// Auth is the SSO authorization for the store
+type Auth struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+	OTP      string `json:"otp"`
+}
+
+// KeyRegister is the request to submit a signing-key to the store
+type KeyRegister struct {
+	Auth
+	AuthorityID string `json:"authority-id"`
+	KeyName     string `json:"key-name"`
+}
+
+// RegisterKey registers an account-key assertion with the store
+func RegisterKey(keyAuth KeyRegister, keypair datastore.Keypair) error {
+
+	// Login to the store as the user
+	// Permissions are documented at https://dashboard.snapcraft.io/docs/api/macaroon.html#request-a-macaroon
+	permissions := []string{"edit_account", "modify_account_key"}
```

Comment:
Any plan to handle these permissions from the UI or are they fix?

Reply:
They are really fixed. These are the minimum that we need for the operations that we are using.